core/src/databases/table.rs

```diff
@@ -993,14 +1028,45 @@ impl Row {
     pub fn row_id(&self) -> &str {
         &self.row_id
     }
-    pub fn content(&self) -> &serde_json::Map<String, serde_json::Value> {
-        &self.value
+    pub fn content(&self) -> serde_json::Map<String, serde_json::Value> {
+        self.value()
+    }
+}
+
+impl Serialize for Row {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: serde::Serializer,
+    {
+        use serde::ser::SerializeStruct;
+        let mut state = serializer.serialize_struct("Row", 3)?;
+        state.serialize_field("row_id", &self.row_id)?;
+        state.serialize_field("value", &self.value())?;
+        // we do not serialize is_delete
+        state.end()
+    }
+}
+
+impl<'de> Deserialize<'de> for Row {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: serde::Deserializer<'de>,
+    {
+        let value = serde_json::Map::deserialize(deserializer)?;
+
+        match value["value"].as_object() {
+            Some(value) => Ok(Row::new_with_value(
+                value["row_id"].to_string(),
```

Comment:
This is a problem, as we won't always have a row_id. What about the scenario where we don't have one but just use the row index as id?

Reply:
I have a repro when using the `/rows` API without a `row_id`.